docs/docs/introduction/getting-started.mdx

```diff
@@ -37,91 +37,121 @@ bower install --save recoil
+You will need to have an App.js file component that you imported to the index.js file.
+The App.js file will take in the following component TexInput, CharCout
 
-## Atom
-
-An **atom** represents a piece of **state**. Atoms can be read from and written to from any component. Components that read the value of an atom are implicitly **subscribed** to that atom, so any atom updates will result in a re-render of all components subscribed to that atom:
+import './App.css';
+import TextInput from './TextInput';
+import CharCout from './CharCout';
 
-```javascript
-const textState = atom({
-  key: 'textState', // unique ID (with respect to other atoms/selectors)
-  default: '', // default value (aka initial value)
-});
-```
 
-Components that need to read from _and_ write to an atom should use `useRecoilState()` as shown below:
+const App = () => {
 
-```jsx
-function CharacterCounter() {
   return (
-    <div>
-      <TextInput />
-      <CharacterCount />
-    </div>
+         <main>
+              <TextInput /> 
+              <CharCout />
+         </main>
+   
   );
 }
 
-function TextInput() {
-  const [text, setText] = useRecoilState(textState);
+export default App;
 
-  const onChange = (event) => {
-    setText(event.target.value);
-  };
+We will then naviagate to our src folder to create Those components
+We'll implement the `CharacterCounter` component in the following section.
 
-  return (
-    <div>
-      <input type="text" value={text} onChange={onChange} />
-      <br />
-      Echo: {text}
-    </div>
-  );
-}
+## Atom
+
+An **atom** represents a piece of **state**. Atoms can be read from and written to from any component. Components that read the value of an atom are implicitly **subscribed** to that atom, so any atom updates will result in a re-render of all components subscribed to that atom:
+
+We would start here with the TextInput component
+
+```javascript
+    import React from 'react'
+    import {atom, useRecoilState} from 'recoil';
+
+    export const inputState = atom({
+        key: "inputState",
+        default: "",
+      });
+
+
+    function TextInput() {
+        const [input, setInput] = useRecoilState(inputState);
+
+        const onChange = (event) => setInput(event.target.value)
+
+      return (
+        <div>
+            <h1>Text Input</h1>
+                <input
+                    type='text' 
+                    value={input}
+                    onChange={onChange}
+                />
+                <br />
+                {'Echo'}
+                <br />
+                {input}
+        </div>
+      )
+    }
+
+
+    export default TextInput
 ```
 
 ## Selector
 
 A **selector** represents a piece of **derived state**. Derived state is a **transformation** of state. You can think of derived state as the output of passing state to a pure function that modifies the given state in some way:
 
+We would then proceed to our CharCount component 
```

Comment:
Same as above to keep the selector and `<CharacterCount>` definitions in separate code blocks.